https://github.com/dajobe/rasqal/pull/13

From b5fdf2539b5eec823ca449de8d04defe37f97ba7 Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Tue, 19 Dec 2023 21:31:36 +0100
Subject: [PATCH] configure: Fix incorrect argument type in gcry_md_open

The gcry_md_open function expects a gcry_md_hd_t * argument,
and not a gcry_md_hd_t argument (which is also a pointer behind
the typedef).  Future compilers will likely treat this as an
error, causing the check to fail unconditionally.
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 113d4f89..6aed2f46 100644
--- a/configure.ac
+++ b/configure.ac
@@ -771,7 +771,7 @@ elif test "$ac_cv_header_gcrypt_h" = "yes"; then
   LIBS="$LIBS `$LIBGCRYPT_CONFIG --libs`"
 
   AC_LINK_IFELSE([AC_LANG_PROGRAM([[
-  #include <gcrypt.h>]], [[ gcry_md_hd_t hash; gcry_md_open(hash, GCRY_MD_MD5, 0); ]])],[have_digest_gcrypt=yes],[have_digest_gcrypt=no])
+  #include <gcrypt.h>]], [[ gcry_md_hd_t hash; gcry_md_open(&hash, GCRY_MD_MD5, 0); ]])],[have_digest_gcrypt=yes],[have_digest_gcrypt=no])
 
   CPPFLAGS="$oCPPFLAGS"
   LIBS="$oLIBS"

